Feat manager prod page

SaadAli2016 opened the pull request:
css issue to look

ahenningsson committed:
Product page - Manager

SaadAli2016 committed:
summaryRequired

SaadAli2016 committed:
Merge branch 'main' into feat-manager-prod-page

SaadAli2016 committed:
css problem

SaadAli2016 committed:
Added missing css file refrence

ahenningsson commented on src/components/productsPage/content.js:
```diff
@@ -1,25 +1,81 @@
+        modal += '</div>';
+        modal += '<div class="modal-footer"></div>'
+        modal += '</div></div>';
+        return modal;
```
Should not be written this way, this is an outdated way of creating elements.

`let modal = $('<div class="modal-content"></div>)`

is the jQuery-way of doing it.

ahenningsson commented on src/components/productsPage/content.js:
```diff
@@ -1,25 +1,81 @@
+        table += '<caption> Bought Items </caption>';
+        table += '<tr><th>Transaction ID</th><th>Admin ID<th>Beer ID</th><th>Amount</th><th>Price</th><th>Time</th><th>Remaining stock</th><th colspan ="2">Actions</th></tr>';
+        
+        var rows = boughtItems.length;
+
+        for (var row = 0; row < rows; ++row)
```
```suggestion
        for (var row = 0; row < boughtItems.length ; ++row)
```

ahenningsson replied:
No reason to create extra variables

ahenningsson commented on src/components/productsPage/content.js:
```diff
@@ -1,25 +1,81 @@
+        for (var row = 0; row < rows; ++row)
+        {
+            let stockquantity = window.getRemainingStock(boughtItems[row].beer_id);
+            table += '<td>' + boughtItems[row].transaction_id + '</td><td>' + boughtItems[row].admin_id + '</td><td>' + boughtItems[row].beer_id +
+                '</td><td>' + boughtItems[row].amount + '</td><td>' + boughtItems[row].price + '</td><td>' + boughtItems[row].timestamp +
+                '</td><td>'+stockquantity+'</td><td><button id = "editButton" type = "button" value = "' + boughtItems[row].transaction_id +
+                '" onclick = edit("' + boughtItems[row].transaction_id + '")>Edit</button></td>';
+                table+='<td><button id = "deleteButton" type = "button" value = "' + boughtItems[row].transaction_id +
+                '" onclick = ondelete("' + boughtItems[row].transaction_id + '")>Delete</button></td>';
+            table += '</tr>';
```
This is not very readible and it should also be made into a function "createRow()" or something

ahenningsson commented on src/css/managerPage.css:
```diff
@@ -0,0 +1,144 @@
+  padding: 5px;
+}
+
+tr:nth-child(even) {
+  background-color: #c3daca;
+}
```
The entire table should follow the same color scheme as the rest of the website imo, and also not be transparent the odd rows.

ahenningsson commented on src/db/database.js:
```diff
@@ -151,20 +178,19 @@ export default class Database {
-                            "alkoholhalt": alkoholhalt,
-                            "forpackning": forpackning
-                            };
+            id: id,
+            namn: namn,
+            pris: pris,
+            varugrupp: varugrupp,
+            land: land,
+            producent: producent,
+            alkoholhalt: alkoholhalt,
+            forpackning: forpackning,
+        };
```
Why is this changed? JSON objects are created in name/value-pairs, and they are surrounded by quotation marks.

ahenningsson commented on src/lang/en.js:
```diff
@@ -56,4 +56,6 @@ export const en = {
   "dropdown-filter": "Filter",
   "dropdown-reset": "Remove filter options",
+
+  "orderNewItemButton" : "+ OrderNewItem",
```
Missing out on the titles in the table

ahenningsson reviewed (commented):
👍

SaadAli2016 committed:
Styling and language Implemented

Changed the css to match with the theme.
Added Translations.
Added confirm box on deletion.
---- one small issue ---
it dosen't get the languge data unless i change the language

SaadAli2016 committed:
Merge branch 'main' into feat-manager-prod-page

ahenningsson reviewed (commented):
great!

SaadAli2016 committed:
Label change and function call to generate strings
